general/chainerrl/baselines/action_wrappers.py

```diff
@@ -0,0 +1,340 @@
+            raise ValueError('Unknown key {} in priorities.'.format(key))
+
+    # Set query
+    query = generate_query(
+        env_name, priorities, reverse_keys, discrete_indices,
+        max_camera_range, num_camera_discretize, allow_pitch)
+
+    return DiscreteActionConverter(query)
+
+
+def generate_continuous_converter(env_name, allow_pitch, max_camera_range):
+    '''
+    Generate a continuous action converter for expert dataset.
+
+    Parameters
+    ----------
+    env_name
+        Environment name.
+    allow_pitch
+        If it is true, it enables a converter to take pitch control action.
+    max_camera_range
+        Maximum value of yaw control.
+    '''
+
+    env_name = env_name.replace("Dense", "")
+
+    def action_converter(action):
+        episode_len = len(action['forward'])
+        value = np.zeros((episode_len, len(action) + 1), dtype=np.float32)
+        idx = 0
+        for key in ALL_ORDERED_KEYS:
+            if key not in action:
+                continue
+            if key in BINARY_KEYS:
+                value[:, idx] = -1 + action[key] * 2
+                idx += 1
+            elif key == 'camera':
+                if allow_pitch:
+                    value[:, idx] = np.clip(action[key][:, 0] / max_camera_range,
+                                            -1, 1)
+                value[:, idx + 1] = np.clip(action[key][:, 1] / max_camera_range,
+                                            -1, 1)
+                idx += 2
+            else:
+                value[:, idx] = -1 + action[key] / (NUM_ENUM_ACTIONS[env_name][key] - 1) * 2
+                idx += 1
+
+        return value
+
+    return action_converter
+
+
+def generate_multi_dimensional_softmax_converter(
+        allow_pitch, max_camera_range, num_camera_discretize):
+    '''
+    Generate a multi dimensional discrete action converter for expert dataset.
+
+    Parameters
+    ----------
+    allow_pitch
+        If it is true, it enables a converter to take pitch control action.
+    num_camera_discretize
+        Number of discretization of yaw control (must be odd).
+    max_camera_range
+        Maximum value of yaw control.
+    '''
+
+    def action_converter(action):
+        episode_len = len(action['forward'])
+        value = np.zeros((episode_len, len(action) + 1), dtype=np.int32)
+        idx = 0
+        for key in ALL_ORDERED_KEYS:
+            if key not in action:
+                continue
+            if key == 'camera':
+                if allow_pitch:
+                    scaled = (
+                        (action[key][:, 0] + max_camera_range)
+                        * (num_camera_discretize - 1)
+                        / (max_camera_range * 2))
+                    value[:, idx] = np.clip(np.round(scaled),
+                                            0, num_camera_discretize - 1)
+                else:
+                    value[:, idx] = np.ones_like(value[:, idx], dtype=np.int32) * (num_camera_discretize - 1) // 2
```

Comment:
this would make you choose the one in the center, right? wouldn' that be different from zero if the number of discretization points is an even number?